### ISSUE TITLE
BePDF does not populate bookmarks treeview when opening a PDF with bookmarks

### ISSUE THREAD

**Issue by grexe** [HaikuArchives/BePDF#115]
When the bookmarks panel is open and I open a PDF with bookmarks, BePDF still only shows an empty treeview.

Only after closing the bookmarks panel and showing it again correctly displays the PDF bookmarks contained in the document.

**Comment by OscarL** [comment]
I was about to open a new ticket, but given that this is already here, I'll just copy/paste my notes here

----

BePDF booksmark bug:

- Open a pdf file.
- Click on the "Show bookmarks" icon on the toolbar.
- Add some user defined bookmark (Menu: Bookmark->Add).
- Close BePDF.

notice that the file now has a "bepdf:bookmarks" attribute (flattened bmessage).

- Reopen the pdf file with BePDF.
- Notice that the bookmarks pane is empty.
- Click once on the "Show bookmarks" icon to hide that pane.
- Click again on the "Show bookmarks" icon to show the bookmarks.
- Notice that the bookmarks pane now shows the previously added bookmark.
- Close BePDF.

That's "part one" of the bug. Bit weird, but OK.

Now the infuriating part... (assume you had spent a good amount of time adding custom bookmarks, not just one for testing)

- Reopen the pdf file with BePDF.
- Notice that the bookmarks pane is empty (boo!).
- Click on the "Show bookmarks" icon.
- Click on the "Show bookmarks" icon again.
- Notice that the bookmarks pane contains all your bookmarks (pheew!).
- Click on the "Show pages list" toolbar icon.
- Click on on the "Show bookmarks" icon.
- Notice that the bookmarks pane is empty.

Now cry... because your bookmarks are gone for good :-(

notice that the file now has lost its "bepdf:bookmarks" attribute.

----

Seems that if the "show bookmarks" is enabled at startup, it misses reading the attributes, and on BePDF closing time... it is not "saved back", and you lose data.

If Show bookmarks is NOT enabled at startup, and you have to click on it to enable it... it works as intended, but do not forget to hide the bookmarks before closing, or you risk losing data the next time you open the file.
